packages/commonwealth/client/scripts/views/components/Header/TOSModal.tsx

```diff
@@ -11,22 +13,32 @@ type TOSModalProps = {
 };
 
 export const TOSModal = ({ onModalClose, onAccept }: TOSModalProps) => {
+  const chain = app.chain ? app.chain.meta : null;
+  const terms = app.chain ? chain.terms : null;
+
   return (
     <div className="TOSModal">
-      <div className="close-button-wrapper">
-        <CWIconButton
-          iconButtonTheme="primary"
-          iconName="close"
-          iconSize="small"
-          className="close-icon"
-          onClick={onModalClose}
-        />
-      </div>
+      <CWModalHeader
+        label="Terms of Service"
+        icon="warning"
+        onModalClose={onModalClose}
+      />
       <div className="content-wrapper">
         <CWText>
-          By clicking accept you agree to the community's Terms of Service
+          By clicking accept you agree to the community's
+          <a href={terms} target="_blank">
+            Terms of Service
+          </a>
+          .
         </CWText>
-        <CWButton onClick={onAccept} label="Accept" />
+      </div>
+      <div className="compact-modal-footer">
```

Comment:
you worked on header, but why not footer? it is duplicated dozens of times as well.

Reply:
Addressed. CWModalFooter added.